central/reports/scheduler/schedule_test.go

```diff
@@ -0,0 +1,96 @@
+
+	To address these findings, please review the impacted software packages in the container images running within deployments you are responsible for and update them to a version containing the fix, if one is available.`
+
+var expectedNoVulnsFoundEmailTemplateRhacsBranding = `
+	Red Hat Advanced Cluster Security for Kubernetes has found zero vulnerabilities associated with the running container images owned by your organization.`
+
+var expectedNoVulnsFoundEmailTemplateStackroxBranding = `
+	StackRox has found zero vulnerabilities associated with the running container images owned by your organization.`
+
+type vulnsAndDate struct {
+	WhichVulns string
+	DateStr    string
+}
+
+func generateExpectedVulnReportEmailTemplates(t *testing.T) (string, string) {
+	data := &vulnsAndDate{
+		WhichVulns: "for all vulnerabilities",
+		DateStr:    time.Now().Format("January 02, 2006"),
+	}
+
+	tmpl, err := template.New("VulnsRHACS").Parse(expectedVulnReportEmailTemplateRhacsBrandingWithPlaceholders)
+	assert.Nil(t, err)
+	var tpl0 bytes.Buffer
+	err = tmpl.Execute(&tpl0, data)
+	assert.Nil(t, err)
+
+	expectedVulnReportEmailTemplateRhacsBranding := tpl0.String()
+
+	tmpl, err = template.New("VulnsStackrox").Parse(expectedVulnReportEmailTemplateStackroxBrandingWithPlaceholders)
+	assert.Nil(t, err)
+	var tpl1 bytes.Buffer
+	err = tmpl.Execute(&tpl1, data)
+	assert.Nil(t, err)
+
+	expectedVulnReportEmailTemplateStackroxBranding := tpl1.String()
+
+	return expectedVulnReportEmailTemplateRhacsBranding, expectedVulnReportEmailTemplateStackroxBranding
+}
+
+func TestVulnMessageBranding1(t *testing.T) {
+	envIsolator := envisolator.NewEnvIsolator(t)
+	rc := fixtures.GetValidReportConfiguration()
+
+	expectedVulnReportEmailTemplateRhacsBranding, expectedVulnReportEmailTemplateStackroxBranding := generateExpectedVulnReportEmailTemplates(t)
+
+	tests := map[string]struct {
+		productBranding string
+		vulnReport      string
+		noVulnReport    string
+	}{
+		"RHACS branding": {
+			productBranding: "RHACS_BRANDING",
+			vulnReport:      expectedVulnReportEmailTemplateRhacsBranding,
+			noVulnReport:    expectedNoVulnsFoundEmailTemplateRhacsBranding,
+		},
+		"StackRox branding": {
+			productBranding: "STACKROX_BRANDING",
+			vulnReport:      expectedVulnReportEmailTemplateStackroxBranding,
+			noVulnReport:    expectedNoVulnsFoundEmailTemplateStackroxBranding,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			envIsolator.Setenv(branding.ProductBranding, tt.productBranding)
+
+			receivedBrandedVulnFound, err := formatMessage(rc, vulnReportEmailTemplate)
+			assert.Nil(t, err)
+			receivedBrandedNoVulnFound, err := formatMessage(rc, noVulnsFoundEmailTemplate)
+			assert.Nil(t, err)
+
+			assert.Equal(t, tt.vulnReport, receivedBrandedVulnFound)
+			assert.Equal(t, tt.noVulnReport, receivedBrandedNoVulnFound)
```

Comment:
Most of my comments may become no-ops and you could simplify the code of this test if you would only check substring presence in the generated reports instead of checking the entire string for equivalence.
Checking string for equivalence forces you to copy the entire template strings which will result in a failed test when the source template string changes.
If the intent of the test is to _check that the branding is properly applied in the output_, you don't need to check the entire output.
See [assert.Contains](https://pkg.go.dev/github.com/stretchr/testify/assert?utm_source=godoc#Contains).

Reply:
It would indeed make things easier as I could scrap the entire `generateExpectedVulnReportEmailTemplates` function. You also have a point that checking whether the email entire email is correct is not strictly within the scope of the changes I made, but on the other hand just checking whether the branding is _somewhere_ inside the email is relatively loose.

Since the correct generation of email texts was previously untested wouldn't it make sense to keep it as is anyways? Better test coverage seems like a net positive to me; even at the price of having to touch that test again when the source template string changes. But here I would defer to the usual way we handle such situations in team maple. @msugakov thoughts?

Reply:
We may not have usual ways for some situations and we're not trying to codify best practices. The downside is that you often need to (re)search how things are done. The upside is that instead of debating definitions and trying to maintain universal unambiguous book of laws, we focus on the problem at hand which usually has its own nuances.

I would say that the entire contents of email and branding of email are related but not the same. For me it would be ok if you want to verify the entire message. In this case, it is better to rename `TestFormatVulnMessageBranding` to just `TestFormatVulnMessage` or `TestFormatMessage` but still iterate over all combinations of branding, etc.

Reply:
In that case I'll leave it in and rename the test as you suggested, since it'd feel wasteful to remove it now. Next time I might write tests that are more narrow in scope.